### PR TITLE
refactor: change consents id put responses to be more explicit

### PR DESCRIPTION
**Pull request by kleyow**
Continuing with the trend of null fields being removed.

PUT /consents/{{ID}} now represented by three responses.

> Auth-Service updating the PISP with a challenge to sign

Represented by ConsentsIDPutResponseUnsigned.yaml

> PISP updating the Auth-Service with the signed challenge

Represented by ConsentsIDPutResponseSigned.yaml

> Auth-Service updating the PISP and DFSP with the finalized credential

Represented by ConsentsIDPutResponseVerified.yaml

### CODE REVIEW

File: docs/thirdparty-openapi3-snippets.yaml [hunk 1]
```diff
@@ -608,16 +608,47 @@ components:
       required:
         - status
         - revokedAt
-    ConsentsIDPutResponse:
-      title: ConsentsIDPutResponse
+    ConsentsIDPutResponseSigned:
```
Comment (eoln) [inline docs/thirdparty-openapi3-snippets.yaml:611]: Signed or Unsigned ?

**Review by eoln — CHANGES_REQUESTED** [review]
- little inconsistency
- a lot of renaming - let see what @lewisdaly would say

File: docs/thirdparty-openapi3-snippets.yaml (hunk 1 above)
Comment (kleyow) [reply docs/thirdparty-openapi3-snippets.yaml:611]: fixed. thanks.

**Review by eoln — APPROVED** [review]
I would wait on @lewisdaly to see his opinion about renaming

File: docs/thirdparty-openapi3-snippets.yaml [hunk 2]
```diff
@@ -608,16 +608,47 @@ components:
...
+      title: ConsentsIDPutResponseSigned
       type: object
       description: |
-        The HTTP request `PUT /consents/{ID}` is used to update a specified consent object.
+        The HTTP request `PUT /consents/{ID}` is used to request a PISP to sign a challenge.
```
Comment (lewisdaly) [inline docs/thirdparty-openapi3-snippets.yaml:615]: I think this should be 
```
The HTTP request `PUT /consents/{ID}` is user by the PISP to update a Consent with a signed challenge and register a credential.
```

File: docs/thirdparty-openapi3-snippets.yaml [hunk 3]
```diff
@@ -636,7 +667,40 @@ components:
...
+        - requestId
+        - participantId
+        - initiatorId
+        - scopes
+        - credential
+    ConsentsIDPutResponseVerified:
+      title: ConsentsIDPutResponseVerified
+      type: object
+      description: |
+        The HTTP request `PUT /consents/{ID}` is used to request a PISP to sign a challenge.
```
Comment (lewisdaly) [inline docs/thirdparty-openapi3-snippets.yaml:681]:
```suggestion
        The HTTP request `PUT /consents/{ID}` is used by the DFSP or Auth-Service to update a Consent object once it has been Verified.
```

File: docs/thirdparty-openapi3-snippets.yaml [hunk 4]
```diff
@@ -937,6 +979,36 @@ components:
...
+          enum:
+            - PENDING
+          description: The challenge has signed but not yet verified.
+        challenge:
+          $ref: '#/components/schemas/CredentialChallengeSigned'
+        payload:
+          type: string
+          description: Base64 encoded bytes - The public key of the Public/Private keypair.
+      required:
+        - type
```
Comment (lewisdaly) [inline docs/thirdparty-openapi3-snippets.yaml:1009]: `id` should be required

**Review by lewisdaly — CHANGES_REQUESTED** [review]
Just a couple minor points. I commented on the compiled file, but hopefully it's not too hard to make the changes on the individual files.

Thanks for this Kevin, I admire your pursuit for good type definitions.

**Comment by kleyow** [comment]
@lewisdaly addressed comments. thank you.

**Review by lewisdaly — APPROVED** [review]

**Pull request merged**